src/virtualship/utils.py

```diff
@@ -42,7 +42,25 @@ def _generic_load_yaml(data: str, model: BaseModel) -> BaseModel:
     return model.model_validate(yaml.safe_load(data))
 
 
-def mfp_to_yaml(excel_file_path: str, yaml_output_path: str):  # noqa: D417
+def load_coordinates(file_path):
+    """Loads the coordinates from a file."""
+    try:
+        return pd.read_excel(file_path)  # Try reading as Excel
+    except ValueError:
+        pass
+
+    try:
+        return pd.read_csv(file_path)  # Try reading as CSV with commas
+    except ValueError:
+        pass
+
+    df = pd.read_csv(file_path, delimiter=";")  # Try reading CSV with colons
+    if len(df.columns) == 1:
+        return None
+    return df
```

Comment:
Since the file_path will contain the extension, isn't is better to just infer the file type by the extension instead of try and error?

Reply:
The idea was that sometimes a file can have a missing extension while still being valid, but I see your points 🙌

I have changed the implementation to check based on the file extension, and also improved the error handling.